src/nvs.rs

```diff
@@ -110,6 +122,19 @@ impl NvsCustom {
 
         Ok(Self(c_partition))
     }
+
+    fn erase(partition: &str) -> Result<(), EspError> {
+        let mut registrations = NONDEFAULT_LOCKED.lock();
+        let c_partition = CString::new(partition).unwrap();
```

Comment:
```suggestion
        let c_partition = CString::new(partition).unwrap();
        let mut registrations = NONDEFAULT_LOCKED.lock();
```

Could switch the lines to reduce time with locked mutex.

Reply:
actually led me to release the lock while erasing since I noticed it might take some time on bigger partitions